src/file/file.grpc.ts

```diff
@@ -40,13 +47,22 @@ export class FileMethods {
    * UpdateFiles updates a list of files and responses with a list of the files' id
    * which succeeded the operation.
    */
-  public static async UpdateFiles(call: ServerUnaryCall<{ partialFile: Partial<IFile>, idList: string[] }>) {
+  public static async UpdateFiles(call: ServerUnaryCall<{ partialFile: Partial<IFile>, idList: string[] }>)
+  : Promise<{ updated: string[], failedFiles: FailedFile[] }> {
     const { updated, failed } = await FileService.updateMany(call.request.idList, call.request.partialFile);
     const traceID = getCurrTraceId();
+    const failedFiles: FailedFile[] = [];
     for (let i = 0; i < failed.length; i++) {
+      let name: string = '';
+      try {
+        const file = await FileService.getById(failed[i].id);
+        name = file.name;
+      } catch (e) {}
       log(Severity.ERROR, failed[i].error.message, 'update files error', traceID, failed[i]);
```

Comment:
`message` of `failed[i].error` can be undefined, use `failed[i].error.toString()` instead.